CommandLine/testing/Platform/TestHarness-X11.cpp

```diff
@@ -174,10 +174,10 @@ int build_game(const string &game, const TestConfig &tc, const string &out) {
   // Invoke the compiler via emake
   using TC = TestConfig;
   string emake_cmd = "./emake";
-  string workdir = (tc.workdir.empty() ? std::string(env_workdir) : tc.workdir);
-  if (!workdir.empty()) workdir = "--workdir=" + workdir;
-  string codegen = (tc.codegen.empty() ? std::string(env_codegen) : tc.codegen);
-  if (!codegen.empty()) codegen = "--codegen=" + codegen;
+  tc.workdir = (tc.workdir.empty() ? std::string(env_workdir) : tc.workdir);
+  string workdir = (tc.workdir.empty() ? std::string("") : ("--workdir=" + workdir));
```

Comment:
Nit/note: the `""` is unnecessary. Actually, you might be able to get away with using it exclusively (not mentioning `string`). Too tired to remember what the standard says about that (thinking it flies), but GCC is generally smarter than that, anyway.

Reply:
That's fine. I thought they'd extended type promotion in ternary to allow constructors when doing POD/non-POD ternary. Anyway, I don't actually care what you do about this line. 😛

Reply:
I decided to remove the argument and just call the default empty string constructor: 1d5eb3d65aa68db86dcede7601ccc373596608be

I hope this doesn't create ambiguity by making it look like a declaration though: https://stackoverflow.com/a/925569